src/reports.cpp

```diff
@@ -602,31 +611,34 @@ static config unit_moves(reports::context & rc, const unit* u)
 		const terrain_type& info = rc.map().get_terrain_info(terrain);
 
 		if (info.union_type().size() == 1 && info.union_type()[0] == info.number() && info.is_nonnull()) {
+			terrain_moves.insert(std::make_pair(info.name(), u->movement_cost(terrain)));
```

Comment:
Please use `std::set::emplace`.

Reply:
We try to always use in-place construction when possible. Just a good modern c++ practice (IMO) practice and keeps the code shorter.